nntrainer/layers/gru.cpp

```diff
@@ -0,0 +1,184 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2020 Jijoong Moon <jijoong.moon@samsung.com>
+ *
+ * @file   gru.cpp
+ * @date   17 March 2021
+ * @brief  This is Gated Recurrent Unit Layer Class of Neural Network
+ * @see    https://github.com/nnstreamer/nntrainer
+ * @author Jijoong Moon <jijoong.moon@samsung.com>
+ * @bug    No known bugs except for NYI items
+ *
+ */
+
+#include <cmath>
+#include <gru.h>
+#include <layer_internal.h>
+#include <lazy_tensor.h>
+#include <nntrainer_error.h>
+#include <nntrainer_log.h>
+#include <parse_util.h>
+#include <util_func.h>
+
+namespace nntrainer {
+
+const std::string GRULayer::type = "gru";
+
+enum GRUParams { weight_xh, weight_hh, bias_h };
+
+#define NUM_GATE 3
+
+// - weight_xh ( input to hidden )
+//  : [1, 1, input_size, unit (hidden_size) x NUM_GATE] -> f, g, i, o
+// - weight_hh ( hidden to hidden )
+//  : [1, 1, unit (hidden_size) , unit (hidden_size) x NUM_GATE] -> f, g, i, o
+// - bias_h ( hidden bias )
+//  : [1, 1, 1, unit (hidden_size) x NUM_GATE] -> f, g, i, o
+int GRULayer::initialize(Manager &manager) {
```

Comment:
can you please consider reutilizing LSTM/RNN code for GRU, and reducing code for all 3. 
code maintenance will be tough if all 3 stay independent.

Reply:
let me think. thanks.